source/common/network/address_impl.cc

```diff
@@ -122,12 +114,13 @@ Ipv4Instance::Ipv4Instance(const std::string& address, uint32_t port,
   ip_.ipv4_.address_.sin_port = htons(port);
   int rc = inet_pton(AF_INET, address.c_str(), &ip_.ipv4_.address_.sin_addr);
   if (1 != rc) {
-    throw EnvoyException(fmt::format("invalid ipv4 address '{}'", address));
+    // Invalid ipv6 address, return.
```

Comment:
I'd make these constructors private and expose static methods of the class that do the validation and construction either prior to or after constructing the object. In this particular case it looks a little more convenient to construct the object first from the private factory method, and then call a new method openSocket() which will do the call to inet_pton and capture the error in the StatusOr if needed.

Maybe in all cases the validation could happen after the construction, since then the data structure is set up to (e.g.) call validateIpv4Supported(friendly_name_).

Reply:
Thanks, will do the refactoring.